libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/RX/NetworkInterface.c

```diff
@@ -127,14 +168,19 @@ BaseType_t xNetworkInterfaceOutput( NetworkBufferDescriptor_t * const pxDescript
     data to be sent as two separate parameters.  The start of the data is located
     by pxDescriptor->pucEthernetBuffer.  The length of the data is located
     by pxDescriptor->xDataLength. */
-    if(0  > SendData( pxDescriptor->pucEthernetBuffer, pxDescriptor->xDataLength ))
+    if( xPHYLinkStatus != 0 )
     {
-        vReleaseNetworkBufferAndDescriptor( pxDescriptor );
-        return pdFALSE;
+        if(SendData( pxDescriptor->pucEthernetBuffer, pxDescriptor->xDataLength ) >= 0)
+        {
+            xReturn = pdTRUE;
+            /* Call the standard trace macro to log the send event. */
+            iptraceNETWORK_INTERFACE_TRANSMIT();
+        }
```

Comment:
The old code always releases in the error case.  Should this behavior be maintained for backwards compatibility?

Reply:
No it should not be maintained, it was incorrect. A network buffer may only be released if `xReleaseAfterSend` is true.